BlogEngine/BlogEngine.Core/Services/Security/Security.cs

```diff
@@ -204,6 +201,18 @@ public static bool AuthenticateUser(string username, string password, bool remem
             return false;
         }
 
+        private static bool IsLocalUrl(string url)
+        {
+            if (string.IsNullOrEmpty(url))
+            {
+                return false;
+            }
+            else
+            {
+                return ((url[0] == '/' && (url.Length == 1 || (url[1] != '/' && url[1] != '\\'))) || (url.Length > 1 && url[0] == '~' && url[0] == '/'));
```

Comment:
There seems to be a condition here that can never be true - this part: (url.Length > 1 && url[0] == '~' && url[0] == '/')